### PR TITLE
MNT: Replace deprecated PIL calls

### PR DESCRIPTION
**Pull request by pllim**
This pull request is to get rid of these warnings in CI that appeared with Pillow 9.1.0 release:

`DeprecationWarning: XXX is deprecated and will be removed in Pillow 10 (2023-07-01). Use Resampling.XXX instead.`

` DeprecationWarning: XXX is deprecated and will be removed in Pillow 10 (2023-07-01). Use Intent.XXX instead.`

### CODE REVIEW

File: ginga/trcalc.py [hunk 1]
```diff
@@ -23,11 +25,18 @@ def use(pkgname):
...
+                      linear=Image.BILINEAR,
+                      area=Image.HAMMING,
+                      bicubic=Image.BICUBIC,
+                      lanczos=Image.LANCZOS)
+else:
+    pil_resize = dict(nearest=Image.Resampling.NEAREST,
+                      linear=Image.Resampling.BILINEAR,
+                      area=Image.Resampling.HAMMING,
+                      bicubic=Image.Resampling.BICUBIC,
+                      lanczos=Image.Resampling.LANCZOS)
```
Comment (ejeschke) [inline ginga/trcalc.py:39]: What do you think about instead, something fairly simple like:
```python
# TODO: can simplify once we no longer support pillow < v9.1
resample = getattr(Image, 'Resampling', Image)
pil_resize = dict(nearest=resample.NEAREST,
                  linear=resample.BILINEAR,
                  area=resample.HAMMING,
                  bicubic=resample.BICUBIC,
                  lanczos=resample.LANCZOS)
```

File: ginga/util/rgb_cms.py [hunk 2]
```diff
@@ -331,10 +332,16 @@ def set_rendering_intent(intent):
...
+    if PIL_LT_9_1:
+        d = dict(absolute_colorimetric=ImageCms.INTENT_ABSOLUTE_COLORIMETRIC,
+                 perceptual=ImageCms.INTENT_PERCEPTUAL,
+                 relative_colorimetric=ImageCms.INTENT_RELATIVE_COLORIMETRIC,
+                 saturation=ImageCms.INTENT_SATURATION)
+    else:
+        d = dict(absolute_colorimetric=ImageCms.Intent.ABSOLUTE_COLORIMETRIC,
+                 perceptual=ImageCms.Intent.PERCEPTUAL,
+                 relative_colorimetric=ImageCms.Intent.RELATIVE_COLORIMETRIC,
+                 saturation=ImageCms.Intent.SATURATION)
```
Comment (ejeschke) [inline ginga/util/rgb_cms.py:344]: Similarly,
```python
    # TODO: can simplify once we no longer support pillow < v9.1
    intent = getattr(ImageCms, 'Intent', ImageCms)
    d = dict(absolute_colorimetric=intent.INTENT_ABSOLUTE_COLORIMETRIC,
             perceptual=intent.INTENT_PERCEPTUAL,
             relative_colorimetric=intent.INTENT_RELATIVE_COLORIMETRIC,
             saturation=intent.INTENT_SATURATION)
```

**Comment by ejeschke** [comment]
I'm also ok with the PR as is if you think it is more clear.

**Comment by pllim** [comment]
@ejeschke , I would prefer to have it as-is to future-proof the code. When your PIL minversion is 9.1 or later, you can easily grep for `PIL_LT_9_1` to get rid of unnecessary checks. Is that okay?

**Comment by ejeschke** [comment]
I'm ok with it, @pllim.

**Comment by pllim** [comment]
Can we merge then? 😸

**Comment by ejeschke** [comment]
FFTM

**Comment by pllim** [comment]
Thanks!

**Pull request merged**